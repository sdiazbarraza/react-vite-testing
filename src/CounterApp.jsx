import {useState} from 'react'
import PropTypes from 'prop-types'

//Lo dejamos afuera evitamos que se cree una funcion cada vez que se renderiza el componente

export const CounterApp = ({value}) => {
    const [counter, setCounter] = useState(value);

    const handleAdd = (e) => {
        setCounter(counter + 1);
    }

    const handleSubstrac = (e) => {
      setCounter(counter - 1);
    }

    const handleReset = (e) => {
      setCounter(value);
    }
  return (
    <>
        <h1>CounterApp</h1>
        <h2>{counter}</h2> 

        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleSubstrac }>-1</button>
        <button onClick={ handleReset }>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}