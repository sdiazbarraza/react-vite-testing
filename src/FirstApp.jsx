import PropTypes from 'prop-types';
export const FirstApp = ({title, subTitle, otro}) => {

  return (
    // React.Fragment: permite agrupar elementos sin necesidad de un div
    <>
     
     <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{otro}</p>
    </>
  )
}
// PropTypes: permite definir el tipo de dato que se espera recibir en las propiedades   
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}
