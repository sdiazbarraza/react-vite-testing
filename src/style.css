html, body {
  font-size: 1.3rem;
  padding: 70 px;
  font-family: 'Roboto', sans-serif;
  background-color: #f1efef;
  color: rgb(20, 12, 12)
}

button{
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}