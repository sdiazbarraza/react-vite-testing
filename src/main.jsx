import React from 'react';
//Importamos para renderizar el componente en web, puede ser en react-native o react-vr
import ReactDOM from 'react-dom/client';

import {App} from './HelloWorldApp';
import {FirstApp} from './FirstApp';
import {CounterApp} from './CounterApp';
import './style.css';
//Renderizamos el componente en el DOM
ReactDOM.createRoot(document.getElementById('root'))
        .render(
            <React.StrictMode>
               <CounterApp value={12}/>
               {/* <FirstApp title="Hola soy german" subTitle={12}/>*/}
            </React.StrictMode>
        );