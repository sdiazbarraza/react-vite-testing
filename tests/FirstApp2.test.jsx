import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en FirstApp', () => {
    const title = "Hola, Soy Goku";
    const subTitle = 12;

    test('debe de hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title="Hola soy german" subTitle={12} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Hola, Soy Goku"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();        
    });

    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subTitle={subTitle} />);

        expect(screen.getAllByText(subTitle).length).toBe(1);
    });
});