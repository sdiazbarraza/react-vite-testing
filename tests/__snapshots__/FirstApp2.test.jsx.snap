// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy german
  </h1>
  <p>
    12
  </p>
  <p />
</div>
`;
